### PR TITLE
Interactive conveniences

### PR DESCRIPTION
**Pull request by ssanderson**

### CODE REVIEW

File: zipline/lib/adjusted_array.pyx [hunk 1]
```diff
@@ -216,10 +227,17 @@ cdef class _Float64AdjustedArrayWindow:
         self.anchor += 1
         return out
 
-    def __repr__(self):
-        return "%s(window_length=%d, anchor=%d, max_anchor=%d)" % (
-            type(self).__name__,
-            self.window_length,
-            self.anchor,
-            self.max_anchor,
+    def inspect(self):
```
Comment (llllllllll) [inline zipline/lib/adjusted_array.pyx]: why is this not the repr?

File: zipline/pipeline/__init__.py [hunk 2]
```diff
@@ -1,16 +1,63 @@
...
+    ----------
+    daily_bar_path : str
+        Path to pass to `BcolzDailyBarReader`.
+    adjustments_path : str
+        Path to pass to SQLiteAdjustmentReader.
+    asset_db_path : str
+        Path to pass to `AssetFinder`.
+    calendar : pd.DatetimeIndex
+        Calendar to use for the loader.
+    warmup_assets : bool
```
Comment (llllllllll) [inline zipline/pipeline/__init__.py]: bool, optional

File: zipline/pipeline/__init__.py [hunk 3]
```diff
@@ -1,16 +1,63 @@
...
+        Whether or not to populate AssetFinder caches.  This can speed up
+        initial latency on subsequent pipeline runs, at the cost of extra
+        memory consumption.
+    """
+    loader = USEquityPricingLoader.from_files(daily_bar_path, adjustments_path)
+
+    if not asset_db_path.startswith("sqlite:"):
```
Comment (llllllllll) [inline zipline/pipeline/__init__.py]: does it need to be sqlite? the backend supports any sqldb no?

File: zipline/pipeline/pipeline.py [hunk 4]
```diff
@@ -145,3 +145,20 @@ def to_graph(self, screen_name, default_screen):
...
 
         return TermGraph(columns)
+
+    def show_graph(self, format='svg'):
+        """
+        Render this Pipeline as a DAG.
+
+        Parameters
+        ----------
+        format : str, optional
```
Comment (llllllllll) [inline zipline/pipeline/pipeline.py]: `{'svg', 'png', 'jpeg'}`

File: zipline/pipeline/pipeline.py [hunk 5]
```diff
@@ -145,3 +145,20 @@ def to_graph(self, screen_name, default_screen):
         columns[screen_name] = screen
 
         return TermGraph(columns)
+
+    def show_graph(self, format='svg'):
```
Comment (llllllllll) [inline zipline/pipeline/pipeline.py]: maybe `expect_value` to check that I don't pass format='ayy'

**Comment by llllllllll** [comment]
Other than the comments looks good.

File: zipline/lib/adjusted_array.pyx (hunk 1 above)
Comment (ssanderson) [inline zipline/lib/adjusted_array.pyx]: I had it as the repr originally, but that's really annoying if you have a list of these or something.

File: zipline/pipeline/__init__.py (hunk 3 above)
Comment (ssanderson) [inline zipline/pipeline/__init__.py]: Maybe this is better as:

`if os.path.isfile(asset_db_path):`?

File: zipline/pipeline/pipeline.py (hunk 5 above)
Comment (ssanderson) [inline zipline/pipeline/pipeline.py]: I added an explicit `elif` and a default branch that raises.

Comment (llllllllll) [inline zipline/pipeline/pipeline.py]: kk

**Pull request merged**